Cost Estimation/BrickWallCostEstimate.FCMacro

```diff
@@ -0,0 +1,144 @@
+        
+        dry_vol_sand = vol_sand*1.333
+        
+        cement_bag = 1.25*(12*25.4)**3
+        num_cement = vol_cement/cement_bag
+
+        # setting up units
+        small_trolley = 75*(12*25.4)**3
+        large_trolley = 325*(12*25.4)**3
+        # This unit is used in India/Pakistan
+        seenkara = 100*(12*25.4)**3
+        truck = 5.5*seenkara
+        big_truck = 12*seenkara 
+        
+        sand_truck = dry_vol_sand/truck
+        sand_big_truck = dry_vol_sand/truck
+        sand_small_trolley = dry_vol_sand/small_trolley
+        sand_large_trolley =dry_vol_sand/large_trolley
+        sand_seenkara =dry_vol_sand/seenkara
+        
+        self.num_bricks += num_bricks
+        self.brick_cost += brick_cost
+        self.dry_vol_sand += dry_vol_sand
+        self.sand_small_trolley += sand_small_trolley
+        self.sand_large_trolley += sand_large_trolley
+        self.sand_truck += sand_truck
+        self.sand_big_truck += sand_big_truck
+        self.sand_seenkara += sand_seenkara 
+        self.num_cement += num_cement
+        self.Length +=  obj.Shape.Length
+        self.Volume += obj.Shape.Volume
+    
+    def all_wall_estimate(self):
+        '''
+        Function to estimate cost of all walls
+        '''
+        for obj in FreeCAD.ActiveDocument.Objects:
+            try:
+                if obj.IfcRole == 'Wall':
+                    self.wall_cost_estimate(obj)
+                elif obj.IfcType == 'Wall':
+                    self.wall_cost_estimate(obj)
+            except AttributeError:
+                pass
+        
+        return vars(self)
+    
+    def selected_wall_estimate(self):
+        '''
+        Function to estimate cost of selected walls
+        Don't use the same object twice with this method
+        '''
+        for obj in FreeCADGui.Selection.getSelection():
+            try:
+                if obj.IfcRole == 'Wall':
+                    self.wall_cost_estimate(obj)
+                elif obj.IfcType == 'Wall':
+                    self.wall_cost_estimate(obj)
+            except AttributeError:
+                pass
+        
+        return vars(self)
+    
+    def pretty_all_wall_estimate(self):
+        '''
+        Function to estimate cost of all walls and prints it in a readable format
+        '''
+        for obj in FreeCAD.ActiveDocument.Objects:
+            try:
+                if obj.IfcRole == 'Wall':
+                    self.wall_cost_estimate(obj)
+                elif obj.IfcType == 'Wall':
+                    self.wall_cost_estimate(obj)
+            except AttributeError:
+                pass
+            
+        print('Number of Bricks :', self.num_bricks)
+        print('Cement Bags: ', self.num_cement)
+        print('Volume of Sand (cu ft):', self.dry_vol_sand*(10**-3*3.28)**3, 'cubic feet')
+
+
+WallEstimator().pretty_all_wall_estimate()
+
+# Use this for cost estimation of a particular wall
+#WallEstimator().selected_wall_estimate()
```

Comment:
Improvement suggestion: make this line available by non-hackers by adding a dialog.